File: trunk/test/exp.swift
class Test
    simd routine bla(real r) -> real result
        result = r
    end

    routine main() -> int result
        simd{real} a = 40000000x
        index i = 0x
        while i < 40000000x
            a[i] = c_call real rand_float()
            i = i + 1x
        end

        c_call start_timer()
        simd i: 0x, 40000000x
            a@ = math::exp(a@)
        end
        c_call stop_timer()

        result = 0
    end
end

File: trunk/test/exp.c
#include <math.h>
#include <stdlib.h>

float rand_float();
void start_timer();
void stop_timer();

int main()
{
    float* a = (float*) malloc(40000000 * sizeof(float));
    for (size_t i = 0; i < 40000000; ++i)
        a[i] = rand_float();

    start_timer();
    for (size_t i = 0; i < 40000000; ++i)
        a[i] = expf(a[i]);
    stop_timer();
}
